### PR TITLE
Code style enhancements and less sphinx warnings

### PR DESCRIPTION
**Pull request by Arusekk**
This is several code style fixes, described in detail in this PR's commit messages.

### CODE REVIEW

File: pwnlib/commandline/asm.py [hunk 1]
```diff
@@ -5,6 +5,9 @@
 import argparse
 import sys
 
+import pwnlib
+pwnlib.args.free_form = False
```
Comment (zachriggle) [inline pwnlib/commandline/asm.py]: Can this part be in a different PR since it's not a code style change?

File: pwnlib/encoders/i386/delta.py [hunk 2]
```diff
@@ -56,26 +55,30 @@ class i386DeltaEncoder(Encoder):
     blacklist  = set(raw)
 
     def __call__(self, raw_bytes, avoid, pcreg=''):
-        table = collections.defaultdict(lambda: [])
+        table = collections.defaultdict(list)
         endchar = bytearray()
 
-        not_bad = lambda x: six.int2byte(x) not in avoid
```
Comment (zachriggle) [inline pwnlib/encoders/i386/delta.py]: Can these encoder fixes go into their own PR?

File: pwnlib/encoders/i386/delta.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def __call__(self, raw_bytes, avoid, pcreg=''):
                 print('No encodings for character %02x' % c)
                 return None
 
-            res.append(choice(choices))
+            res += choice(choices)
```
Comment (zachriggle) [inline pwnlib/encoders/i386/delta.py]: Can all of the encoder changes go into their own PR?

Comment (Arusekk) [reply pwnlib/encoders/i386/delta.py]: Yup, here it is: #1583

File: pwnlib/commandline/asm.py (hunk 1 above)
Comment (Arusekk) [reply pwnlib/commandline/asm.py]: Yup, moved to #1573, which is all about CLI

File: pwnlib/elf/corefile.py [hunk 4]
```diff
@@ -551,15 +551,15 @@ def __init__(self, *a, **kw):
...
 
-        if not self.elftype == 'CORE':
+        if self.elftype != 'CORE':
             log.error("%s is not a valid corefile" % self.file.name)
 
-        if not self.arch in prstatus_types.keys():
+        if self.arch not in prstatus_types:
             log.warn_once("%s does not use a supported corefile architecture, registers are unavailable" % self.file.name)
 
-        prstatus_type = prstatus_types.get(self.arch, None)
```
Comment (heapcrash) [inline pwnlib/elf/corefile.py:560]: Today I learned!

File: pwnlib/elf/elf.py [hunk 5]
```diff
@@ -464,9 +464,6 @@ def _describe(self, *a, **kw):
             self.checksec(*a, **kw)
         )
 
-    def __repr__(self):
```
Comment (heapcrash) [inline pwnlib/elf/elf.py:467]: Can we have this back?  Thanks!

File: pwnlib/tubes/remote.py [hunk 6]
```diff
@@ -99,7 +99,7 @@ def _connect(self, fam, typ):
 
                 sock = socket.socket(self.family, self.type, self.proto)
 
-                if timeout != None and timeout <= 0:
+                if timeout is not None and timeout <= 0:
```
Comment (heapcrash) [inline pwnlib/tubes/remote.py:102]: Overall I feel like these `!= None` and `is not None` changes are overkill and just introduce code churn but other than gut feeling I don't see anything wrong with it.

File: pwnlib/ui.py [hunk 7]
```diff
@@ -97,11 +97,11 @@ def yesno(prompt, default=None):
         cur = default
         while True:
             k = term.key.get()
-            if   k in ('y', 'Y', '<left>') and cur != True:
+            if   k in ('y', 'Y', '<left>') and cur is not True:
```
Comment (heapcrash) [inline pwnlib/ui.py:100]: This should likely be `not cur` or similar.  We should generally not check for `is True` or `is False`.

**Review by heapcrash — CHANGES_REQUESTED** [review]

File: pwnlib/elf/elf.py (hunk 5 above)
Comment (Arusekk) [reply pwnlib/elf/elf.py:467]: It is already there (the function was defined twice 🤣)

File: pwnlib/ui.py (hunk 7 above)
Comment (heapcrash) [reply pwnlib/ui.py:100]: Ping @Arusekk

Comment (Arusekk) [reply pwnlib/ui.py:100]: Here the value can be either of three: `True`, `False` or `None`. You are probably right, if you prefer so, you can commit this further change.
